app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsEmailAddressController.scala

```diff
@@ -77,11 +92,97 @@ class ContactDetailsEmailAddressController @Inject() (
 
   private def updateRegistration(
     formData: EmailAddress
-  )(implicit req: JourneyRequest[AnyContent]): Future[Either[ServiceError, Registration]] =
-    update { registration =>
-      val updatedEmailAddress =
-        registration.primaryContactDetails.copy(email = Some(formData.value))
-      registration.copy(primaryContactDetails = updatedEmailAddress)
+  )(implicit request: JourneyRequest[AnyContent]): Future[Either[ServiceError, Registration]] =
+    request.user.identityData.credentials.map { creds =>
+      emailVerificationConnector.getStatus(creds.providerId).flatMap {
+        case Right(emailStatusResponse) =>
+          emailStatusResponse match {
+            case Some(response) =>
+              update { registration =>
+                registration.copy(primaryContactDetails =
+                                    updatedPrimaryContactDetails(formData, registration),
+                                  metaData = registration.metaData.add(response.emails)
+                )
+              }
+            case None =>
+              update { registration =>
+                registration.copy(primaryContactDetails =
+                  updatedPrimaryContactDetails(formData, registration)
+                )
+              }
+          }
+        case Left(error) => throw error
+      }
+    }.getOrElse(
+      Future(
+        Left(
+          DownstreamServiceError("Cannot find user credentials id",
+                                 RegistrationException("Cannot find user credentials id")
+          )
+        )
+      )
+    )
+
+  private def updatedPrimaryContactDetails(formData: EmailAddress, registration: Registration) =
+    registration.primaryContactDetails.copy(email = Some(formData.value))
+
+  private def updatedJourneyId(registration: Registration, journeyId: String)(implicit
+    request: JourneyRequest[AnyContent]
+  ): Future[Either[ServiceError, Registration]] = {
+    val primaryContact = registration.primaryContactDetails.copy(journeyId = Some(journeyId))
+    update {
+      registration => registration.copy(primaryContactDetails = primaryContact)
     }
+  }
+
+  private def saveAndContinue(registration: Registration, email: String)(implicit
+    hc: HeaderCarrier,
+    request: JourneyRequest[AnyContent]
+  ): Future[Result] =
+    if (appConfig.emailVerificationEnabled)
+      EmailVerificationService.getPrimaryEmailStatus(registration) match {
+        case VERIFIED =>
+          Future(Redirect(routes.ContactDetailsTelephoneNumberController.displayPage()))
+        case NOT_VERIFIED =>
+          request.user.identityData.credentials.map { creds =>
+            createEmailVerification(creds.providerId,
+                                    registration.primaryContactDetails.email.get
```

Comment:
This could potentially have an exception being raised here, If email is `None`

Reply:
Fixed